assembly.py

```diff
@@ -93,120 +96,110 @@ def trim_rmdup_subsamp_reads(inBam, clipDb, outBam, n_reads=100000):
         read_utils.purge_unmated(rmdupfq[0], rmdupfq[1], purgefq[0], purgefq[1])
-        did_purge_unmated = True
         n_purge = util.file.count_fastq_reads(purgefq[0])
         if n_purge < n_reads:
             log.warn("We purged %s reads down to %s mated reads, which is below subsample threshold (%s). TO DO: add smarter subsampling in this scenario. Proceeding with %s mated reads for now." %(
                 n_rmdup, n_purge, n_reads, n_purge))
 
-    # Log count
-    log.info("PRE-SUBSAMPLE COUNT: %s read pairs", n_purge)
-
-    # Subsample to a reasonable input size for Trinity
-    subsampfq = list(map(util.file.mkstempfname,
-                         ['.subsamp.1.fastq', '.subsamp.2.fastq']))
-    subsampfq_unpaired = list(map(util.file.mkstempfname,
-                                  ['.subsamp.unpaired.1.fastq', '.subsamp.unpaired.2.fastq']))
-    if n_purge <= n_reads:
-        log.info("skipping subsampling of reads: read count (%s) <= subsample threshold (%s)" %(
-            n_purge, n_reads))
-        if did_purge_unmated:
-            for i in range(2):
-                shutil.copyfile(purgefq[i], subsampfq[i])
-        else:
-            for i in range(2):
-                shutil.copyfile(rmdupfq[i], subsampfq[i])
-                shutil.copyfile(rmdupfq_unpaired[i], subsampfq_unpaired[i])
-        n_subsamp = n_purge
-    else:
-        # We should have purged unmated since n_purge > n_reads ==>
-        # n_rmdup > n_reads, so only work with purgefq reads
-        assert did_purge_unmated
-        # TODO: Move subsampling to after the fastq -> BAM conversion, and
-        # use 'samtools view -s' or Picard's DownsampleSam
-        cmd = [os.path.join(util.file.get_scripts_path(), 'subsampler.py'),
-               '-n',
-               str(n_reads),
-               '-mode',
-               'p',
-               '-in',
-               purgefq[0],
-               purgefq[1],
-               '-out',
-               subsampfq[0],
-               subsampfq[1],]
-        util.misc.run_and_print(cmd, check=True)
-        n_subsamp = util.file.count_fastq_reads(subsampfq[0])
-
-    # Fastq -> BAM
-    # Note: this destroys RG IDs! We should instead frun the BAM->fastq step in a way
-    # breaks out the read groups and perform the above steps in a way that preserves
-    # the RG IDs.
     tmp_header = util.file.mkstempfname('.header.sam')
     tools.samtools.SamtoolsTool().dumpHeader(inBam, tmp_header)
-    if n_subsamp == 0:
-        # FastqToSam cannot deal with empty input
-        # but Picard SamFormatConverter can deal with empty files
-        opts = ['INPUT=' + tmp_header, 'OUTPUT=' + outBam, 'VERBOSITY=ERROR']
-        tools.picard.PicardTools().execute('SamFormatConverter', opts, JVMmemory='50m')
+
+    # convert paired reads to bam
+    # stub out an empty file if the input fastqs are empty
+    tmp_bam_paired = util.file.mkstempfname('.paired.bam')
+    if all(util.file.count_fastq_reads(x) > 0 for x in  purgefq) > 0:
```

Comment:
I'm not sure I understand the if expression here, there appears to be an extraneous `> 0` at the end. Should it be this?
```
if all(util.file.count_fastq_reads(x) > 0 for x in purgefq)
```

Also, if you're just testing for emptiness, can you just use os.path.getsize or similar? That should be O(1) instead of O(n) on the input.